### PR TITLE
Provide a way to build Pooling options based on protocol version

### PR DESCRIPTION
**Pull request by jorgebay**

### CODE REVIEW

File: src/Cassandra.IntegrationTests/Core/PoolShortTests.cs [hunk 1]
```diff
@@ -214,5 +214,40 @@ public void Connect_With_Ssl_Test()
...
+                TestHelper.WaitUntil(() =>
+                    pool.OpenConnections == coreConnectionLength);
+                var ports = sCluster.GetConnectedPorts();
+                //coreConnectionLength + 1 (the control connection) 
+                Assert.AreEqual(coreConnectionLength + 1, ports.Count);
+                
+                TestHelper.WaitUntil(() =>
+                    {
+                        //Force pool to resize connections to max
+                        pool.ConsiderResizingPool(int.MaxValue);
```
Comment (jorgebay) [inline src/Cassandra.IntegrationTests/Core/PoolShortTests.cs]: We should avoid using internal methods in integration tests.
Using core connections to test that the settings are being honored is enough for me.

File: src/Cassandra.Tests/BuilderTests.cs [hunk 2]
```diff
@@ -184,23 +184,18 @@ public void WithMaxProtocolVersion_Validates_Greater_Than_Zero()
         }
 
         [Test]
-        public void PoolingOptions_Create_Based_On_Protocol_Version()
+        [TestCase(ProtocolVersion.MaxSupported, 1, 2)]
+        [TestCase(ProtocolVersion.V2, 2, 8)]
+        public void PoolingOptions_Create_Based_On_Protocol_Version(ProtocolVersion protocolVersion,
```
Comment (jorgebay) [inline src/Cassandra.Tests/BuilderTests.cs]: +1 for the refactoring.

**Comment by landim** [comment]
Removed the max connection check on integration (avoiding to have to call internal method)

**Comment by jorgebay** [comment]
This needs to be rebased in order to work, I'll do it.

**Comment by jorgebay** [comment]
Test fixes look good to me.
I've rebased it and squashed @landim  commits into 1.

**Pull request merged**